machine/machine_core/ssh_connection.py

```diff
@@ -328,19 +338,19 @@ def execute(self, command=None, script=None, input=None, environment={},
                             proc.stdout.close()
                         else:
                             if self.verbose:
-                                os.write(sys.__stdout__.fileno(), data)
+                                write_all(sys.__stdout__.fileno(), data)
                             output += data.decode('utf-8', 'replace')
                     elif fd == stderr_fd:
                         data = os.read(fd, 1024)
                         if not data:
                             rset.remove(stderr_fd)
                             proc.stderr.close()
                         elif not quiet or self.verbose:
-                            os.write(sys.__stderr__.fileno(), data)
+                            write_all(sys.__stderr__.fileno(), data)
                 for fd in ret[1]:
                     if fd == stdin_fd:
                         if input:
-                            num = os.write(fd, input.encode('utf-8'))
+                            num = write_all(fd, input.encode('utf-8'))
```

Comment:
This should not be replaced, as we do expect it to have partial writes.

Reply:
Argh, of course. I fixed it to just retry on BlockingIOError.